static/src/javascripts/projects/facia/modules/ui/container-show-more.js

```diff
@@ -2,77 +2,168 @@ define([
             type: 'session'
         }) || {};
-        if (state !== 'displayed') {
+        if (state !== STATE_DISPLAYED) {
             delete prefs[containerId];
         } else {
             prefs[containerId] = 'more';
         }
-        userPrefs.set(prefName, prefs, {
+        userPrefs.set(PREF_NAME, prefs, {
             type: 'session'
         });
     }
 
     function readPrefs(containerId) {
-        var prefs = userPrefs.get(prefName, {
+        var prefs = userPrefs.get(PREF_NAME, {
             type: 'session'
         });
-        return (prefs && prefs[containerId]) ? 'displayed' : 'hidden';
+        return (prefs && prefs[containerId]) ? STATE_DISPLAYED : STATE_HIDDEN;
     }
 
-    function showMore($container, button) {
+    function showMore(button) {
         fastdom.write(function () {
             /**
              * Do not remove: it should retain context for the click stream module, which recurses upwards through
              * DOM nodes.
              */
-            $container.toggleClass(className, button.state === 'displayed');
-            button.state = (button.state === 'hidden') ? 'displayed' : 'hidden';
-            setButtonState(button, button.state);
+            setButtonState(button, (button.state === STATE_HIDDEN) ? STATE_DISPLAYED : STATE_HIDDEN);
             updatePref(button.id, button.state);
         });
     }
 
-    function renderToDom($container, button) {
+    function renderToDom(button) {
         fastdom.write(function () {
-            $container.addClass(className)
+            button.$container.addClass(HIDDEN_CLASS_NAME)
                 .removeClass('js-container--fc-show-more')
-                .toggleClass(className, button.state === 'hidden');
+                .toggleClass(HIDDEN_CLASS_NAME, button.state === STATE_HIDDEN);
             // Initialise state, as it might be different from what was rendered server side based on localstorage prefs
             setButtonState(button, button.state);
         });
     }
 
+    function loadShowMore(pageId, containerId) {
+        return ajax({
+            url: '/' + pageId + '/show-more/' + containerId + '.json',
+            crossOrigin: true,
+            timeout: REQUEST_TIMEOUT
+        });
+    }
+
+    function dedupShowMore($container, html) {
+        var seenArticles = itemsByArticleId($container),
+            $html = bonzo.create(html);
+
+        $(ITEM_SELECTOR, $html).each(function (article) {
+            var $article = bonzo(article);
+            if ($article.attr(ARTICLE_ID_ATTRIBUTE) in seenArticles) {
+                $article.remove();
+            }
+        });
+
+        return $html;
+    }
+
+    function loadShowMoreForContainer(button) {
+        fastdom.write(function () {
+            setButtonState(button, STATE_LOADING);
+        });
+
+        loadShowMore(config.page.pageId, button.id).then(function (response) {
+            var dedupedShowMore = dedupShowMore(button.$container, response.html);
```

Comment:
No, you're right, I forgot about that. We ought to display some kind of message in that case.